src/ImageSharp/Formats/Png/Filters/AverageFilter.cs

```diff
@@ -44,23 +52,34 @@ public static void Decode(byte[] scanline, byte[] previousScanline, int bytesPer
         /// <param name="scanline">The scanline to encode</param>
         /// <param name="previousScanline">The previous scanline.</param>
         /// <param name="result">The filtered scanline result.</param>
+        /// <param name="bytesPerScanline">The number of bytes per scanline</param>
         /// <param name="bytesPerPixel">The bytes per pixel.</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static void Encode(byte[] scanline, byte[] previousScanline, byte[] result, int bytesPerPixel)
+        public static void Encode(BufferSpan<byte> scanline, BufferSpan<byte> previousScanline, BufferSpan<byte> result, int bytesPerScanline, int bytesPerPixel)
```

Comment:
There is no need to pass `bytesPerScanline` as a 3rd parameter, if the `scanline` span had the lenght `bytesPerScanline`.
Can I drop this in into this PR? I would also replace `this.scanLine` and `this.previousScanline` with `Buffer<T>`.